buf breaking in pre-commit will set core.bare to 'true' and failed

### GitHub repository with your minimal reproducible example (do not leave this field blank or fill out this field with "github.com/bufbuild/buf" or we will automatically close your issue, see the instructions above!)

_No response_

### What's up?

When I run `pre-commit run --all-files` or `buf breaking --against "ssh://git@github.com/provistaai/protobuf.git#branch=main"`, It's all right. But when I run `git commit`, It will auto set core.bare to 'true' and failed.

![Image](https://github.com/user-attachments/assets/105d5679-ef86-43e4-9830-9efc2084c7df)

It's a submodule and my `.pre-commit-config.yaml` is:
```yaml
repos:
  - repo: local
    hooks:
      - id: buf-format
        name: Format Protobuf Files
        entry: buf format -w
        language: system
        files: \.proto$
        pass_filenames: false

      - id: buf-breaking
        name: Check Protobuf Breaking Changes
        entry: buf breaking --against "ssh://git@github.com/provistaai/protobuf.git#branch=main"
        language: system
        files: \.proto$
        pass_filenames: false
```
Is that a bug? What should I do? Thx a lot

It's fine to just run `pre-commit`. The problem is in `git commit`:

```sh
PS D:\py\test-a\app\protos> pre-commit
Format Protobuf Files....................................................Passed
- exit code: 1

Failure: could not clone ssh://git@github.com:Wu-Jiayang/test-b.git: exit status 128
ssh: Could not resolve hostname github.com:Wu-Jiayang: Name or service not known
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

PS D:\py\test-a\app\protos> pre-commit
[ERROR] Your pre-commit configuration is unstaged.
`git add .pre-commit-config.yaml` to fix this.
PS D:\py\test-a\app\protos> git add .
PS D:\py\test-a\app\protos> pre-commit
Format Protobuf Files....................................................Passed
Check Protobuf Breaking Changes..........................................Failed
- hook id: buf-breaking
- exit code: 1

Failure: Module "path: "."" had no .proto files

PS D:\py\test-a\app\protos> git commit -m "test"
Format Protobuf Files....................................................Passed
Check Protobuf Breaking Changes..........................................Failed
- hook id: buf-breaking
- exit code: 1

Failure: could not clone https://github.com/Wu-Jiayang/test-b.git: exit status 3
warning: core.bare and core.worktree do not make sense
error: remote origin already exists.
```

--- a/private/pkg/git/cloner.go
+++ b/private/pkg/git/cloner.go
@@ -76,16 +76,12 @@ func (c *cloner) CloneToBucket(
 
 	depthArg := strconv.Itoa(int(depth))
 
-	// In the case where this is being run in an environment where GIT_DIR is set, e.g. within
-	// a submodule, we want to treat this as a stand-alone clone rather than interacting with
-	// an existing GIT_DIR. So we filter out GIT_DIR from our environment variables.
-	filteredEnv := make(map[string]string)
-	envContainer.ForEachEnv(func(key, value string) {
-		if key != "GIT_DIR" {
-			filteredEnv[key] = value
-		}
+	envContainer = app.NewEnvContainerWithOverrides(envContainer, map[string]string{
+		// In the case where this is being run in an environment where GIT_DIR is set, e.g. within
+		// a submodule, we want to treat this as a stand-alone clone rather than interacting with
+		// an existing GIT_DIR. So we filter out GIT_DIR from our environment variables.
+		"GIT_DIR": "",
 	})
-	envContainer = app.NewEnvContainer(filteredEnv)
 
 	baseDir, err := tmp.NewDir(ctx)
 	if err != nil {

--- a/private/pkg/git/cloner.go
+++ b/private/pkg/git/cloner.go
@@ -76,6 +76,17 @@ func (c *cloner) CloneToBucket(
 
 	depthArg := strconv.Itoa(int(depth))
 
+	// In the case where this is being run in an environment where GIT_DIR is set, e.g. within
+	// a submodule, we want to treat this as a stand-alone clone rather than interacting with
+	// an existing GIT_DIR. So we filter out GIT_DIR from our environment variables.
+	filteredEnv := make(map[string]string)
+	envContainer.ForEachEnv(func(key, value string) {
+		if key != "GIT_DIR" {
+			filteredEnv[key] = value
+		}
+	})
+	envContainer = app.NewEnvContainer(filteredEnv)
+
 	baseDir, err := tmp.NewDir(ctx)
 	if err != nil {
 		return err

